streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -152,6 +182,41 @@ public final void removeAllCacheLevelSensors(final String taskName, final String
         }
     }
 
+    public final Sensor storeLevelSensor(final String taskName,
+                                         final String storeName,
+                                         final String sensorName,
+                                         final Sensor.RecordingLevel recordingLevel,
+                                         final Sensor... parents) {
+        final String key = threadName + "." + taskName + ".store." + storeName;
+        synchronized (storeLevelSensors) {
+            if (!storeLevelSensors.containsKey(key)) {
+                storeLevelSensors.put(key, new LinkedList<>());
+            }
+
+            final String fullSensorName = key + "." + sensorName;
+
+            final Sensor sensor = metrics.sensor(fullSensorName, recordingLevel, parents);
+
+            storeLevelSensors.get(key).push(fullSensorName);
+
+            return sensor;
+        }
+    }
+
+    // TODO: should we use this method when the state store gets closed?
```

Comment:
should we? (we don't currently unload any sensors when we close a store, although all the children of a task-level sensor get unloaded with the task)

Reply:
We should, and I think it was actually a bug that we did not do that before.